meson.build

```diff
@@ -0,0 +1,468 @@
+    if libheif_dep.version().version_compare('>=1.7.0')
+        cfg_var.set('HAVE_HEIF_AVIF', '1')
+    endif
+    if get_option('heif-module')
+        cfg_var.set('HEIF_MODULE', '1')
+    endif
+endif
+
+libjxl_dep = dependency('libjxl', version: '>=0.5', required: get_option('jpeg-xl'))
+if libjxl_dep.found()
+    libvips_deps += libjxl_dep
+    libvips_deps += dependency('libjxl_threads', version: '>=0.5')
+    cfg_var.set('HAVE_LIBJXL', '1')
+    if cc.has_function('JxlEncoderInitBasicInfo', prefix: '#include <jxl/encode.h>', dependencies: libjxl_dep)
+        cfg_var.set('HAVE_LIBJXL_JXLENCODERINITBASICINFO', '1')
+    endif
+    if get_option('jpeg-xl-module')
+        cfg_var.set('LIBJXL_MODULE', '1')
+    endif
+endif
+
+libpoppler_dep = dependency('poppler-glib', version: '>=0.16.0', required: get_option('poppler'))
+if libpoppler_dep.found() and pdfium_dep.found()
+    message('PDFium has been found, ignoring Poppler support')
+elif libpoppler_dep.found()
+    libvips_deps += libpoppler_dep
+    libvips_deps += dependency('cairo', version: '>=1.2')
+    cfg_var.set('HAVE_POPPLER', '1')
+    if get_option('poppler-module')
+        cfg_var.set('POPPLER_MODULE', '1')
+    endif
+endif
+
+libnifti_dep = cc.find_library('nifti2', has_headers: 'nifti1_io.h', required: false)
+if not libnifti_dep.found()
+    libnifti_dep = dependency('NIFTI', method : 'cmake', modules : ['NIFTI'], required: get_option('nifti'))
+endif
+if libnifti_dep.found()
+    libvips_deps += libnifti_dep
+    cfg_var.set('HAVE_NIFTI', '1')
+endif
+
+# option('heif-module', type : 'boolean', value : false, description: 'Build libheif as module')
+# option('jpeg-xl-module', type : 'boolean', value : false, description: 'Build libjxl as module')
+# option('magick-module', type : 'boolean', value : false, description: 'Build libMagic as module')
+# option('openslide-module', type : 'boolean', value : false, description: 'Build OpenSlide as module')
+# option('poppler-module', type : 'boolean', value : false, description: 'Build poppler as module')
+
+if cc.has_header('sys/file.h')
+    cfg_var.set('HAVE_SYS_FILE_H', '1')
+endif
+if cc.has_header('sys/param.h')
+    cfg_var.set('HAVE_SYS_PARAM_H', '1')
+endif
+if cc.has_header('sys/mman.h')
+    cfg_var.set('HAVE_SYS_MMAN_H', '1')
+endif
+if cc.has_header('unistd.h')
+    cfg_var.set('HAVE_UNISTD_H', '1')
+endif
+if cc.has_header('io.h')
+    cfg_var.set('HAVE_IO_H', '1')
+endif
+if cc.has_header('direct.h')
+    cfg_var.set('HAVE_DIRECT_H', '1')
+endif
+if cc.has_header('windows.h')
+    cfg_var.set('HAVE_WINDOWS_H', '1')
+endif
+if get_option('deprecated')
+    cfg_var.set('ENABLE_DEPRECATED', '1')
+endif
+if get_option('nsgif')
+    cfg_var.set('HAVE_NSGIF', '1')
+endif
+if get_option('ppm')
+    cfg_var.set('HAVE_PPM', '1')
+endif
+if get_option('analyze')
+    cfg_var.set('HAVE_ANALYZE', '1')
+endif
+if get_option('deprecated')
+    cfg_var.set('ENABLE_DEPRECATED', '1')
+endif
+
```

Comment:
I think we need an option for radiance as well, perhaps:

```
if get_option('radiance')
    cfg_var.set('HAVE_RADIANCE', '1')
endif
```

`radiance` is already in `meson_options.txt`.